Debug MSVC build error

Has anyone see a compile error similar to the following?

```
[1/24] Building CXX object src\polygon\CMakeFiles\polygon.dir\src\polygon.cpp.obj
FAILED: src/polygon/CMakeFiles/polygon.dir/src/polygon.cpp.obj 
C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1440~1.338\bin\Hostx64\x64\cl.exe  /nologo /TP -Dpolygon_EXPORTS -ID:\a\bext\bext\build\manifold\MANIFOLD_BLD-prefix\src\MANIFOLD_BLD\src\polygon\include -ID:\a\bext\bext\build\manifold\MANIFOLD_BLD-prefix\src\MANIFOLD_BLD\src\utilities\include -ID:\a\bext\bext\build\manifold\MANIFOLD_BLD-prefix\src\MANIFOLD_BLD\src\utilities\..\third_party\glm -ID:\a\bext\bext\build\manifold\MANIFOLD_BLD-prefix\src\MANIFOLD_BLD\src\utilities\..\third_party\thrust -ID:\a\bext\bext\build\manifold\MANIFOLD_BLD-prefix\src\MANIFOLD_BLD\src\utilities\..\third_party\thrust\dependencies\libcudacxx\include -ID:\a\bext\bext\build\manifold\MANIFOLD_BLD-prefix\src\MANIFOLD_BLD\src\collider\include /DWIN32 /D_WINDOWS /GR /EHsc /Zi /Ob0 /Od /RTC1 -std:c++17 -MDd /DNOMINMAX /bigobj -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CPP -DMANIFOLD_EXCEPTIONS=1 /showIncludes /Fosrc\polygon\CMakeFiles\polygon.dir\src\polygon.cpp.obj /Fdsrc\polygon\CMakeFiles\polygon.dir\ /FS -c D:\a\bext\bext\build\manifold\MANIFOLD_BLD-prefix\src\MANIFOLD_BLD\src\polygon\src\polygon.cpp
src\utilities\include\vec.h(236): error C2607: static assertion failed
src\utilities\include\vec.h(236): note: the template instantiation context (the oldest one first) is
src\polygon\src\polygon.cpp(306): note: see reference to class template instantiation 'manifold::Vec<`anonymous-namespace'::EarClip::VertItr>' being compiled
```

--- a/src/polygon/src/polygon.cpp
+++ b/src/polygon/src/polygon.cpp
@@ -14,7 +14,6 @@
 
 #include "polygon.h"
 
-#include <algorithm>
 #include <map>
 #include <set>
 
@@ -302,7 +301,7 @@ class EarClip {
 
   struct IdxCollider {
     Collider collider;
-    Vec<VertItr> itr;
+    std::vector<VertItr> itr;
   };
 
   // A circularly-linked list representing the polygon(s) that still need to be
@@ -829,7 +828,7 @@ class EarClip {
   IdxCollider VertCollider(VertItr start) const {
     Vec<Box> vertBox;
     Vec<uint32_t> vertMorton;
-    Vec<VertItr> itr;
+    std::vector<VertItr> itr;
     const Box box(glm::vec3(bBox_.min, 0), glm::vec3(bBox_.max, 0));
 
     Loop(start, [&vertBox, &vertMorton, &itr, &box, this](VertItr v) {

--- a/src/utilities/include/utils.h
+++ b/src/utilities/include/utils.h
@@ -80,6 +80,13 @@ void Permute(Vec<T>& inOut, const Vec<T1>& new2Old) {
   gather(new2Old.begin(), new2Old.end(), tmp.begin(), inOut.begin());
 }
 
+template <typename T, typename T1>
+void Permute(std::vector<T>& inOut, const Vec<T1>& new2Old) {
+  std::vector<T> tmp(std::move(inOut));
+  inOut.resize(new2Old.size());
+  gather(new2Old.begin(), new2Old.end(), tmp.begin(), inOut.begin());
+}
+
 template <typename T>
 T AtomicAdd(T& target, T add) {
   std::atomic<T>& tar = reinterpret_cast<std::atomic<T>&>(target);

